crates/ruff_python_formatter/src/options.rs

```diff
@@ -1,5 +1,32 @@
 use ruff_formatter::printer::{LineEnding, PrinterOptions};
 use ruff_formatter::{FormatOptions, IndentStyle, LineWidth};
+use std::ffi::OsStr;
+use std::path::Path;
+
+#[derive(Clone, Debug)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub enum SourceType {
+    /// A `.py` file
+    Py,
+    /// A `.pyi` file
+    Pyi,
+}
+
+impl SourceType {
+    pub fn from_path(path: &Path) -> Option<Self> {
+        match path.extension().and_then(OsStr::to_str) {
+            Some("py") => Some(Self::Py),
+            Some("pyi") => Some(Self::Pyi),
+            _ => None,
+        }
+    }
+}
+
+impl Default for SourceType {
+    fn default() -> Self {
+        Self::Py
+    }
+}
```

Comment:
https://github.com/astral-sh/ruff/pull/5552/ introduces a new [`PySourceType`](https://github.com/astral-sh/ruff/blob/dhruv/jupyter-cell-magic/crates/ruff/src/source_kind.rs). Would it be possible to move the new type to `ruff_python_ast` and re-use it here?